controllers/deployment.go

```diff
@@ -0,0 +1,160 @@
+		container.Image = application.Spec.Image
+		container.ImagePullPolicy = corev1.PullAlways
+		container.Command = application.Spec.Command
+
+		var uid int64 = 150 // TODO: 65534? Evnt. hashed? Random?
+		deployment.Spec.Template.Spec.SecurityContext = &corev1.PodSecurityContext{}
+		deployment.Spec.Template.Spec.SecurityContext.SupplementalGroups = []int64{uid}
+		deployment.Spec.Template.Spec.SecurityContext.FSGroup = &uid
+
+		yes := true
+		no := false
+		container.SecurityContext = &corev1.SecurityContext{}
+		container.SecurityContext.SeccompProfile = &corev1.SeccompProfile{}
+		container.SecurityContext.SeccompProfile.Type = "RuntimeDefault"
+		container.SecurityContext.Privileged = &no
+		container.SecurityContext.AllowPrivilegeEscalation = &no
+		container.SecurityContext.ReadOnlyRootFilesystem = &yes
+		container.SecurityContext.RunAsUser = &uid
+		container.SecurityContext.RunAsGroup = &uid
+
+		container.Ports = make([]corev1.ContainerPort, 1)
+		container.Ports[0].ContainerPort = int32(application.Spec.Port)
+
+		container.Env = application.Spec.Env
+		container.EnvFrom = make([]corev1.EnvFromSource, len(application.Spec.EnvFrom))
+		envFrom := container.EnvFrom
+
+		for i, env := range application.Spec.EnvFrom {
+			if len(env.ConfigMap) > 0 {
+				envFrom[i].ConfigMapRef = &corev1.ConfigMapEnvSource{}
+				envFrom[i].ConfigMapRef.LocalObjectReference.Name = env.ConfigMap
+			} else if len(env.Secret) > 0 {
+				envFrom[i].SecretRef = &corev1.SecretEnvSource{}
+				envFrom[i].SecretRef.LocalObjectReference.Name = env.Secret
+			}
+		}
+
+		container.Resources = application.Spec.Resources
+
+		deployment.Spec.Template.Spec.Volumes = make([]corev1.Volume, len(application.Spec.FilesFrom)+1)
+		container.VolumeMounts = make([]corev1.VolumeMount, len(application.Spec.FilesFrom)+1)
+		volumes := deployment.Spec.Template.Spec.Volumes
+		volumeMounts := container.VolumeMounts
+
+		volumes[0] = corev1.Volume{Name: "tmp", VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}}
+		volumeMounts[0] = corev1.VolumeMount{Name: "tmp", MountPath: "/tmp"}
+
+		for i, file := range application.Spec.FilesFrom {
+			if len(file.ConfigMap) > 0 {
+				volumes[i+1].Name = file.ConfigMap
+				volumes[i+1].ConfigMap = &corev1.ConfigMapVolumeSource{}
+				volumes[i+1].ConfigMap.Name = file.ConfigMap
+			} else if len(file.Secret) > 0 {
+				volumes[i+1].Name = file.PersistentVolumeClaim
+				volumes[i+1].Secret = &corev1.SecretVolumeSource{}
+				volumes[i+1].Secret.SecretName = file.Secret
+			} else if len(file.PersistentVolumeClaim) > 0 {
+				volumes[i+1].Name = file.PersistentVolumeClaim
+				volumes[i+1].PersistentVolumeClaim = &corev1.PersistentVolumeClaimVolumeSource{}
+				volumes[i+1].PersistentVolumeClaim.ClaimName = file.PersistentVolumeClaim
+			}
+
+			volumeMounts[i+1] = corev1.VolumeMount{Name: volumes[i+1].Name, MountPath: file.MountPath}
+		}
+
+		if application.Spec.Readiness != nil {
```

Comment:
Probes kan ikke endres i kubernetes API-et, er det noen måte å hindre at de prøver å endre det? kubebuilder "annotasjon" eller noe sånt for å definere det i CRD-en?